doc/guides/device_mgmt/smp_groups/smp_group_0.rst

```diff
@@ -0,0 +1,258 @@
+    +---------------+-----------------------------------------------+
+    | ``5``         | System reset                                  |
+    +---------------+-----------------------------------------------+
+
+Echo command
+============
+
+Echo command responses by sending back string that it has received.
+
+Echo request
+------------
+
+Echo request header fields:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``0``        |  ``0``         |
+    +--------+--------------+----------------+
+    | ``2``  | ``0``        |  ``0``         |
+    +--------+--------------+----------------+
+
+CBOR Payload of request command:
+
+.. code-block:: none
+
+    {
+        (str)"d" : (str)
+    }
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "d"                   | string to be replied by echo service              |
+    +-----------------------+---------------------------------------------------+
+
+Echo response
+-------------
+
+Echo response header fields
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+----------------------------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` | Note                             |
+    +========+==============+================+==================================+
+    | ``1``  | ``0``        |  ``0``         | When request ``OP`` was ``0``    |
+    +--------+--------------+----------------+----------------------------------+
+    | ``3``  | ``0``        |  ``0``         | When request ``OP`` was ``2``    |
+    +--------+--------------+----------------+----------------------------------+
+
+CBOR Payload of response command:
+
+.. code-block:: none
+
+    {
+        (str)"r"        : (str)
+        (str)"rc"       : (int)
+    }
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "r"                   | Replying echo string                              |
+    +-----------------------+---------------------------------------------------+
+    | "rc"                  | :ref:`mcumgr_smp_protocol_status_codes`           |
+    +-----------------------+---------------------------------------------------+
+
+Task statistics command
+=======================
+
+The command responds with some system statistics.
+
+Task statistics request
+-----------------------
+
+Task statistics request header fields
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``0``        |  ``2``         |
+    +--------+--------------+----------------+
+
+The command does not send any payload in request, or sends empty CBOR map.
```

Comment:
Thinking about it, probably best to clarify this, I have been wondering for some time now why we can't use our SMP commands without a dummy payload so just had a look and in the `mgmt_ctxt_init` function it's initialising the cbor parser/encoder, so if you don't send any payload the packet is binned. I would personally regard it as a bug because it just seems pointless to me to send an empty cbor array if the command doesn't need any cbor data to run, but should be noted here that the empty cbor array is a requirement to run a command

Reply:
Done